lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataFactory.php

```diff
@@ -223,7 +223,7 @@ protected function doLoadMetadata($class, $parent, $rootEntityFound, array $nonS
      */
     protected function validateIdentifier(ClassMetadata $class) : void
     {
-        if (! $class->identifier && ! $class->isMappedSuperclass && ! $class->isEmbeddedDocument && ! $class->isQueryResultDocument) {
+        if (! $class->identifier && ! $class->isMappedSuperclass && ! $class->isEmbeddedDocument && ! $class->isQueryResultDocument && ! $class->isView()) {
```

Comment:
It might be time to consider using linebreaks :P
```suggestion
        if (! $class->identifier &&
            ! $class->isMappedSuperclass &&
            ! $class->isEmbeddedDocument &&
            ! $class->isQueryResultDocument &&
            ! $class->isView()) {
```

Reply:
😅